tasks/main.yml

```diff
@@ -46,96 +82,139 @@
-        state: "{{ 'absent' if kernel_settings_transparent_hugepages_defrag ==
-                   __kernel_settings_state_absent else none }}"
-    bootloader:
-      - name: "{{ 'cmdline'
-                  if kernel_settings_bootloader_cmdline | d({})
-                  else none }}"
-        value: "{{ kernel_settings_bootloader_cmdline |
-                   d([]) }}"
-    purge: "{{ kernel_settings_purge }}"
-    ansible_managed_new: "{{ __kernel_settings_new_header }}"
-    ansible_managed_current: "{{ __kernel_settings_cur_header }}"
-  notify: __kernel_settings_handler_modified
-  register: __kernel_settings_register_module
-  when:
-    - kernel_settings_sysctl or kernel_settings_sysfs
-      or kernel_settings_systemd_cpu_affinity
-      or kernel_settings_transparent_hugepages
-      or kernel_settings_transparent_hugepages_defrag
-      or kernel_settings_bootloader_cmdline | d({})
-      or kernel_settings_purge
-      or __kernel_settings_cur_header != __kernel_settings_new_header
+  template:
+    src: "{{ __kernel_settings_profile_src }}.j2"
+    dest: "{{ __kernel_settings_profile_filename }}"
+    mode: "0644"
   vars:
-    __kernel_settings_cur_header: "{{
-      __kernel_settings_profile_contents_b64.content | b64decode |
-      regex_replace('(?sm)^\\[.*$', '') }}"
-    __kernel_settings_new_header: "{{
-      lookup('template', 'get_ansible_managed.j2') }}"
+    __sysctl_old: "{{ __kernel_settings_profile_contents.data.get('sysctl', {})
+      if not kernel_settings_purge
+      and kernel_settings_sysctl != __kernel_settings_state_empty
+      and not __kernel_settings_previous_replaced in kernel_settings_sysctl
+      else {} }}"
+    __sysfs_old: "{{ __kernel_settings_profile_contents.data.get('sysfs', {})
+      if not kernel_settings_purge
+      and kernel_settings_sysfs != __kernel_settings_state_empty
+      and not __kernel_settings_previous_replaced in kernel_settings_sysfs
+      else {} }}"
+    __systemd_old: "{{
+      __kernel_settings_profile_contents.data.get('systemd', {}).get('cpu_affinity')
+      if not kernel_settings_purge
+      and kernel_settings_systemd_cpu_affinity != __kernel_settings_state_absent
+      else none }}"
+    __trans_huge_old: "{{
+      __kernel_settings_profile_contents.data.get('vm', {}).get('transparent_hugepages')
+      if not kernel_settings_purge
+      and kernel_settings_transparent_hugepages != __kernel_settings_state_absent
+      else none }}"
+    __trans_defrag_old: "{{
+      __kernel_settings_profile_contents.data.get('vm', {}).get('transparent_hugepage.defrag')
+      if not kernel_settings_purge
+      and kernel_settings_transparent_hugepages_defrag != __kernel_settings_state_absent
+      else none }}"
+    __sections:
+      - name: sysctl
+        new: "{{ kernel_settings_sysctl | difference([__kernel_settings_previous_replaced]) | list
+          if kernel_settings_sysctl != __kernel_settings_state_empty
+          else [] }}"
```

Comment:
Why __sysctl_old is set to dict `{}` and new is set to list `[]`?

Reply:
`new` is a `list` of `dict` - basically the same format as passed in by `kernel_settings_sysctl` et. al. - see https://github.com/linux-system-roles/kernel_settings/pull/220/files#diff-d5995b3130f3bd6a8890b9832f7250dbaf05507afd6e49f7e371152a459c50abR8

`old` is a `dict` as returned by `kernel_settings_get_config` and processed here - https://github.com/linux-system-roles/kernel_settings/pull/220/files#diff-d5995b3130f3bd6a8890b9832f7250dbaf05507afd6e49f7e371152a459c50abR15